src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -85,12 +113,41 @@ public async Task ExecuteAsync(ReindexJobRecord reindexJobRecord, WeakETag weakE
 
                     _reindexJobRecord.Resources.AddRange(resourceList);
                     _reindexJobRecord.SearchParams.AddRange(notYetIndexedParams.Select(p => p.Name));
-                }
 
-                // This is just a shell for now, will be completed in future
-                await CompleteJobAsync(OperationStatus.Completed, cancellationToken);
+                    // generate and run first query
+                    var queryStatus = new ReindexJobQueryStatus(null);
+                    queryStatus.LastModified = DateTimeOffset.UtcNow;
+                    queryStatus.Status = OperationStatus.Queued;
+
+                    _reindexJobRecord.QueryList.Add(queryStatus);
+
+                    // update the complete total
+                    var countOnlyResults = await ExecuteReindexQueryAsync(queryStatus, true, cancellationToken);
```

Comment:
nit: please call out the param name explicitly for non-descriptive values (like null, true, etc). Makes it easier to read.